src/meili/routing.cc

```diff
@@ -38,7 +38,8 @@ bool LabelSet::put(const baldr::GraphId& nodeid, const baldr::GraphId& edgeid,
   // Create a new label and push it to the queue
   if (it == node_status_.end()) {
     const uint32_t idx = labels_.size();
-    if (sortcost < max_cost_) {
+    // We only add the labels if we are under the limits for distance and for time or time limit is 0
+    if (cost.cost < max_dist_ && (max_time_ == 0.f || cost.secs < max_time_)) {
```

Comment:
this is the important part, we dont let the path expand if the distance or time is unreasonable